runtime-modules/storage/src/lib.rs

```diff
@@ -2815,6 +2815,19 @@ impl<T: Trait> Module<T> {
         }
     }
 
+    fn ensure_upload_and_deletion_prize_account_are_coherent(
+        deletion_prize: &Option<DynamicBagDeletionPrize<T>>,
+        params: &UploadParameters<T>,
+    ) -> DispatchResult {
+        ensure!(deletion_prize.is_some(), Error::<T>::AccountsNotCoherent,);
```

Comment:
This check is too strict. It renders useless `Option` of the deletion_prize parameter which contradicts the interface. I was wrong - let's delete this check and allow passing None as dynamic bag deletion prize having only check for a non-empty deletion_prize_account_id.

Reply:
fixed in fe90c57a04